src/main/java/net/spring/service/UserServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package net.spring.service;
+
+import lombok.RequiredArgsConstructor;
+import net.spring.entity.user.Role;
+import net.spring.entity.user.User;
+import net.spring.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final RoleService roleService;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+    private static final String ROLE = "USER";
+
+    @Override
+    public User findByUserName(String userName) {
+        return userRepository.findByUserName(userName).orElse(null);
```

Comment:
or else better throw custom exception with message for example "User with this username not found"